lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2752,108 +2752,20 @@ void glow::fold(Function *F, CompilationMode mode) {
-  DCE().run(F);
-
-  // Sink transpose operations in an attempt to cancel them out.
-  // Perform code sinking until a fixed-point is reached.
-  // On big functions, the number of iterations until the fixpoint
-  // is usually at most 2 or 3 iterations.
-  while (SinkCode().run(F)) {
-    // Perform Dead Code Elimination between rounds of code sinking.
-    DCE().run(F);
-    VLOG_IF_EVERY_N(0, google::COUNTER > 1, 100)
-        << "Warning: sinkCode optimization applied another 100 iterations "
-           "without reaching fixed point";
-  }
-
-  // Transposes that don't move data are optimized into Reshapes, which enables
-  // further optimizations.
-  OptimizeTransposeIntoReshape().run(F);
-  // Need to remove old uses that would prohibit Reshape(Constant) optimization.
-  DCE().run(F);
-
-  // Reshapes and transposes can prevent other optimizations from triggering,
-  // so try to optimize them out first.
-  OptimizeReshape().run(F);
-  if (cctx.compMode == CompilationMode::Infer) {
-    TransposeConstants().run(F);
-  }
-
-  // Perform Common Subexpression Elimination.
-  CSE().run(F);
-
-  // Optimize Pad nodes
-  MergePadIntoConvolution().run(F);
-
-  // Perform Dead Code Elimination.
-  DCE().run(F);
-
-  // Merge multiple matmul nodes into a single large matmul.
-  MergeMatMul().run(F);
-
-  // Merge multiple batched adds into a larger batched add.
-  MergeBatchedAdd().run(F);
-
-  // Merge ReduceMean into AveragePool if possible.
-  OptimizeReduceMean().run(F);
-
-  // Convert BatchMatMuls with a broadcasted RHS to a single MatMul.
-  ConvertBroadcastedBatchMatMul().run(F);
-
-  // Perform Dead Code Elimination.
-  DCE().run(F);
-
-  if (cctx.compMode == CompilationMode::Infer) {
-    // Merge batch normalization operations.
-    // Do after transpose constant folding, as weight transposes can prevent
-    // the optimization from triggering.
-    OptimizeBatchNorm().run(F);
-  }
-
-  // Perform Common Subexpression Elimination.
-  CSE().run(F);
-
-  // Optimize Concat nodes.
-  OptimizeConcatNodes().run(F);
-
-  // Optimize arithmetic nodes based on algebraic identities.
-  OptimizeArithmeticNodes().run(F);
-
-  // Optimize Tensor shape transformations.
-  OptimizeSliceOfSplat().run(F);
-
-  // Merge Transpose into MatMul/FC.
-  // Run DCE to ensure correct number of node users.
-  DCE().run(F);
-  MergeTransposeIntoMatMulOrFC().run(F);
-
-  // Optimize away intermediate type conversions.
-  OptimizeConversions().run(F);
-
-  // Optimize quantization related operators.
-  while (OptimizeQuantization().run(F) || SinkRescaleQuantizedNode().run(F)) {
-    DCE().run(F);
-    VLOG_IF_EVERY_N(0, google::COUNTER > 1, 100)
-        << "Warning: Quantization optimization applied another 100 iterations "
-           "without reaching fixed point";
-  }
-
-  // Optimize reshapes introduced during above optimizations.
-  OptimizeReshape().run(F);
+  FunctionPassManager FPM("TargetIndependentGraphOptzFPM",
```

Comment:
This place looks funny: We ask a target backend to provide the optimization pipeline for target independent optimizations? ;-)

Reply:
May be the name of the pass manager is a copy & paste error?